File: backend/app/models/feature.rb
class Feature < ApplicationRecord
  include ActiveModel::Serializers::JSON

  has_many :comments

  def as_json
    {
      'id' => id,
      'type' => 'feature',
      'attributes' => {
        'external_id' => external_id,
        'magnitude' => magnitude,
        'place' => place,
        'time' => time,
        'tsunami' => tsunami,
        'mag_type' => mag_type,
        'title' => title,
        'coordinates' => {
           'longitude' => longitude,
           'latitude' => latitude,
        },
      },
      'links' => {
        'external_url' => external_url,
      }
    }
  end
end

File: frontend/src/types/Feature.ts
interface FeatureCoordinates {
  longitude: string
  latitude: string
}

interface FeatureAttributes {
  external_id: string
  magnitude: string
  place: string
  time: string
  tsunami: boolean
  mag_type: string
  title: string
  coordinates: FeatureCoordinates
}

interface FeatureLinks {
  external_url: string
}

export default interface Feature {
  id: number
  type: string
  attributes: FeatureAttributes
  links: FeatureLinks
}
